src/npcmove.cpp

```diff
@@ -2237,7 +2255,19 @@ void npc::move_to( const tripoint &pt, bool no_bashing, std::set<tripoint> *nomo
     if( moved ) {
         const tripoint old_pos = pos();
         setpos( p );
-
+        if( old_pos.x - p.x < 0 ) {
+            facing = FD_RIGHT;
+        } else {
+            facing = FD_LEFT;
+        }
+        if( is_mounted() && mounted_creature ) {
+            if( mounted_creature->pos() != pos() ) {
+                mounted_creature->setpos( pos() );
+                mounted_creature->facing = facing;
+                mounted_creature->process_triggers();
```

Comment:
Your code is sometimes very inconsistent: here you use `mounted_creature` several times, but above, you store the pointer in a separate variable and use that several times. Is there any reasoning for when you use this or the other style?